File: final_dataset_A/generated_code/chatgpt/python/Schedule_medical_appointment_017_Secure.py
appointments = []

VALID_PATIENTS = {
    "P1001",
    "P1002"
}

VALID_DOCTORS = {
    "D2001",
    "D2002"
}


def schedule_medical_appointment(
        patient_id,
        doctor_id,
        date,
        time):

    """
    Securely schedules an appointment.
    Validates IDs and checks slot availability.
    """

    if patient_id not in VALID_PATIENTS:
        return "Invalid patient ID."

    if doctor_id not in VALID_DOCTORS:
        return "Invalid doctor ID."

    for appointment in appointments:
        if (
            appointment["doctor_id"] == doctor_id
            and appointment["date"] == date
            and appointment["time"] == time
        ):
            return "Time slot is unavailable."

    appointments.append({
        "patient_id": patient_id,
        "doctor_id": doctor_id,
        "date": date,
        "time": time
    })

    return "Appointment booked successfully."


if __name__ == "__main__":
    print(schedule_medical_appointment(
        "P1001",
        "D2001",
        "2026-08-15",
        "10:30"
    ))

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/java-chatgpt-A/Schedule_medical_appointment_017_Secure.java
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.HashSet;
import java.util.Set;

public class Schedule_medical_appointment_017_Secure {

    private static final Set<String> validPatients = Set.of("P1001", "P1002");
    private static final Set<String> validDoctors = Set.of("D2001", "D2002");
    private static final Set<String> bookedSlots = new HashSet<>();

    public static String scheduleAppointment(
            String patientId,
            String doctorId,
            LocalDate date,
            LocalTime time) {

        if (!validPatients.contains(patientId))
            return "Invalid patient ID.";

        if (!validDoctors.contains(doctorId))
            return "Invalid doctor ID.";

        String slot = doctorId + "-" + date + "-" + time;

        if (bookedSlots.contains(slot))
            return "Selected time slot is unavailable.";

        bookedSlots.add(slot);

        return "Appointment booked successfully.";
    }

    public static void main(String[] args) {
        System.out.println(scheduleAppointment(
                "P1001",
                "D2001",
                LocalDate.now(),
                LocalTime.of(9, 0)));
    }
}